test/controllers/membership_controller_test.rb

```diff
@@ -0,0 +1,60 @@
+require 'test_helper'
+
+class MembershipControllerTest < ActionController::TestCase
+  setup do
+    login_as_user(:babhamozo_leader)
+    request.env['HTTP_REFERER'] = "http://test.host/groups/1"
+  end
+
+  test "successful new member in group" do
+    login_as_user(:non_babhamozo_member)
+    assert_difference('Membership.count', 1) do
```

Comment:
If your business logic is not in the controller then you can also test it properly and does not have to resort to hacks like this.